packages/ui-test-queries/src/utils/firstOrNull.ts

```diff
@@ -21,6 +21,7 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-export function firstOrNull(result) {
+
+export function firstOrNull<T>(result: T | T[] | null): T | null {
   return Array.isArray(result) ? result[0] || null : null
```

Comment:
another code that when properly typed would just result in tons of errors because of the missing null checks. If we refactor this we should do something like 'firstOrError'